### PR TITLE
Fix weekly-docs-check

### LINKED ISSUE #1843: Weekly docs check fails
[Lots of errors](https://github.com/opendp/opendp/actions/runs/10024103114/job/27705772741) like:
```
/home/runner/work/opendp/opendp/python/src/opendp/mod.py:docstring of opendp.mod:5: WARNING: py:mod reference target not found: opendp.domains
/home/runner/work/opendp/opendp/python/src/opendp/mod.py:docstring of opendp.mod.Domain:4: WARNING: py:mod reference target not found: opendp.domains
/home/runner/work/opendp/opendp/python/src/opendp/mod.py:docstring of opendp.mod.Measure:4: WARNING: py:mod reference target not found: opendp.measures
```
Maybe similar to 
- #1829

... generated files not checked in, nor built when CI needs them?

### PR DESCRIPTION
**Pull request by mccalluc**
- Fix #1843

Temporarily enabled this workflow on PRs: Once we've confirmed that it works, switch back to just running as a cronjob. Draft.

### CODE REVIEW

File: docs/linkchecker-external.cfg [hunk 1]
```diff
@@ -20,6 +18,8 @@ ignore=
   # Binder is returning 500s.
   https://mybinder.org/
 
+
+
```
Comment (mccalluc) [inline docs/linkchecker-external.cfg:22]:
```suggestion
```

**Review by mccalluc — COMMENTED** [review]
Self-reviewing, I think there's a redundant cargo build. (Will need one more commit after this to restore this to just being a cronjob.)

File: .github/workflows/weekly-doc-check.yml [hunk 2]
```diff
@@ -10,7 +10,55 @@ on:
...
+          key: ${{ runner.os }}-cargo-${{ hashFiles('rust/Cargo.toml') }}
+          restore-keys: ${{ runner.os }}-cargo-
+
+      - name: Check Rust version
+        run: cargo --version
+
+      - name: Build
+        # We only need "bindings", but build fails without "untrusted".
+        # TODO: All combos of features should work: https://github.com/opendp/opendp/issues/1343
+        run: cargo build --verbose --features bindings,untrusted
```
Comment (Shoeboxam) [inline .github/workflows/weekly-doc-check.yml:52]: To only generate bindings, then run `cargo check` instead.

File: docs/linkchecker-external.cfg [hunk 3]
```diff
@@ -1,6 +1,4 @@
 [filtering]
-checkextern=1
```
Comment (Shoeboxam) [inline docs/linkchecker-external.cfg:2]: Why drop external links?

**Review by Shoeboxam — CHANGES_REQUESTED** [review]

File: .github/workflows/weekly-doc-check.yml (hunk 2 above)
Comment (mccalluc) [reply .github/workflows/weekly-doc-check.yml]: Maybe I'm missing something, but that [doesn't seem to work by itself](https://github.com/opendp/opendp/actions/runs/10180763679/job/28159554880#step:7:189). Are there additional arguments?

Comment (Shoeboxam) [reply .github/workflows/weekly-doc-check.yml]: I was thinking to just substitute `build` with `check`! It still needs features set.

File: docs/linkchecker-external.cfg (hunk 3 above)
Comment (mccalluc) [reply docs/linkchecker-external.cfg:2]: My mistake! Restoring... and I also need to make other changes.

Comment (mccalluc) [reply docs/linkchecker-external.cfg:2]: For comparison:

smoke-test:
```
linkchecker -f linkchecker.cfg build/html/index.html
...
That's it. 468 links in 468 URLs checked. 0 warnings found. 0 errors found.
Stopped checking at 2024-07-31 17:52:22-004 (3 seconds)
```

weekly html check (with latest changes):
```
$ linkchecker --no-warnings -f linkchecker-external.cfg build/html/index.html
...
That's it. 1431 links in 1463 URLs checked. 0 warnings found (6 ignored or duplicates not printed). 0 errors found.
Stopped checking at 2024-07-31 18:13:00-004 (1 minute, 41 seconds)
```

weekly md check:
```
$ find .. | grep '\.md$' | grep -v 'venv/\|.git/\|CHANGELOG\|vendor/\|R/opendp/README.md' | xargs linkchecker --no-warnings -f linkchecker-external.cfg
...
That's it. 71 links in 74 URLs checked. 0 warnings found (19 ignored or duplicates not printed). 0 errors found.
Stopped checking at 2024-07-31 18:18:19-004 (1 minute, 1 seconds)
```

File: .github/workflows/weekly-doc-check.yml (hunk 2 above)
Comment (mccalluc) [reply .github/workflows/weekly-doc-check.yml:51]: (Restored `cargo build` in the latest commit, but if there's something better, great!)

**Review by Shoeboxam — APPROVED** [review]
Will be nice to have these fixed.

**Pull request merged**